aten/src/ATen/native/DispatchStub.h

```diff
@@ -109,12 +109,33 @@ struct RegisterDispatch {
 
 #define DEFINE_DISPATCH(name) struct name name
 
-#if defined(__CUDACC__)
-#define REGISTER_DISPATCH(name, fn) \
+#define REGISTER_ARCH_DISPATCH(name, arch, fn) \
```

Comment:
What is going on that requires all these extra definitions here?

Reply:
This is just some weirdness in how the dispatch stub works due to fbcode requirements. It basically declares static symbols for all CPU archs that are present at compile time, because it assumes we will compile kernels for those for sure. Those kernels are then responsible for filling in missing symbols, and doing so prevents them form being stripped away even without `--whole-archive`.